build.gradle

```diff
@@ -743,61 +743,71 @@ subprojects {
-                            args += [
-                                    "-javacoptions:-ApermitMissingJdk"
-                            ]
-                        } else if (project.name.is('checker')) {
-                            args += [
-                                    "-javacoptions:-classpath ${sourceSets.testannotations.output.asPath}",
-                            ]
-                        }
+                      } else {
+                        args += [
+                            // checker/jtreg/nullness/defaultsPersist/ReferenceInfoUtil.java
+                            // uses the jdk.jdeps module.
+                            "-javacoptions:--add-modules jdk.jdeps",
+                            "-javacoptions:--add-exports=jdk.jdeps/com.sun.tools.classfile=ALL-UNNAMED",
+                            "-vmoptions:--add-opens=jdk.jdeps/com.sun.tools.classfile=ALL-UNNAMED",
+                            "-vmoptions:--add-opens=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED",
+                            "-vmoptions:--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED",
+                            "-vmoptions:--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED",
+                            "-vmoptions:--add-opens=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED",
+                            "-vmoptions:--add-opens=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED",
+                            "-vmoptions:--add-opens=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED",
+                            "-vmoptions:--add-opens=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED",
+                            "-vmoptions:--add-opens=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED",
+                            "-vmoptions:--add-opens=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED",
+                        ]
+                      }
+                      if (project.name.is('framework')) {
+                        // Do not check for the annotated JDK
+                        args += [
+                            "-javacoptions:-ApermitMissingJdk"
+                        ]
+                      } else if (project.name.is('checker')) {
+                        args += [
+                            "-javacoptions:-classpath ${sourceSets.testannotations.output.asPath}",
+                        ]
+                      }
 
-                        // Location of jtreg tests
-                        args += "${tests}"
+                      // Location of jtreg tests
+                      args += "${tests}"
+                    }
+                  } catch (Exception ex) {
+                    if (ex.getCause() != null && ex.getCause().getCause()!= null) {
+                      String msg = ex.getCause().getLocalizedMessage()
+                      msg += ":\n"+ex.getCause().getCause().getLocalizedMessage()
+                      msg += "\nHave you installed jtreg?"
```

Comment:
The error message looks like this:
```
> Task :framework:jtregTests FAILED
Could not start 'jtreg':
Cannot run program "jtreg" (in directory "/Users/smillst/jsr308/checker-framework"): error=2, No such file or directory
Have you installed jtreg?

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':framework:jtregTests'.
> A problem occurred starting process 'command 'jtreg''
```

Should I add something about how to install it? Or what version is required?

Reply:
That error message looks fine.  Thanks!